### PR TITLE
[BUG]: Delete card crash the app

### LINKED ISSUE #380: [BUG]: Deleting a card is crashing live version
## Description

If you delete any card from a board i will crash

Screenshot 2022-07-28 at 10 11 34

### PR DESCRIPTION
**Pull request by mourabraz**
Relates to #380 

## Proposed Changes

  - Fixes the bug when a user delete a card or a card item
  - create a hook to deal with the creation of Socket IO
  - page for editing a "board" splitted in different components

Mention people who discussed this issue previously
@f-morgado

This pull request closes #380

### CODE REVIEW

File: frontend/src/pages/boards/[boardId].tsx [hunk 1]
```diff
@@ -109,233 +91,86 @@ const Board: NextPage<Props> = ({ boardId, mainBoardId }) => {
...
+			  ]
+			: [false, false];
+	}, [board, userId]);
+
+	// Socket IO
+	const [socketId, cleanSocket] = useSocketIO(boardId);
+	useEffect(() => cleanSocket, [cleanSocket]);
+
+	// Show button in sub boards to merge into main
+	const showButtonToMerge = !!(board?.isSubBoard && !board.submitedByUser && isResponsible);
```
Comment (joaosantos99) [inline frontend/src/pages/boards/[boardId].tsx:122]: Everything is looking good to me in the frontend. I just have a little comment, you are transforming in booleon a value from an operation that only returns a booleon.

File: frontend/src/pages/boards/[boardId].tsx [hunk 2]
```diff
@@ -109,233 +91,86 @@ const Board: NextPage<Props> = ({ boardId, mainBoardId }) => {
...
-			</>
-		);
-	}
-	return <LoadingPage />;
-};
-
-Board.defaultProps = {
-	mainBoardId: undefined
+	// Show Alert message if sub board was merged
+	const showMessageIfMerged = !!(board?.submitedByUser && board.submitedAt && mainBoardId);
```
Comment (joaosantos99) [inline frontend/src/pages/boards/[boardId].tsx:139]: Same here about transforming the boolean value

**Review by joaosantos99 — APPROVED** [review]

File: frontend/src/pages/boards/[boardId].tsx (hunk 1 above)
Comment (mourabraz) [reply frontend/src/pages/boards/[boardId].tsx:122]: Because in case `board`  be `undefined`  the statement `board?.isSubBoard` will return `undefined`

Comment (joaosantos99) [reply frontend/src/pages/boards/[boardId].tsx:122]: It makes sense. LGTM 🚀

**Pull request merged**